stacker/providers/aws/default.py

```diff
@@ -1,20 +1,185 @@
+            Attributes={
+                "Policy": self.create_policy()},
+        )
+
+        result = self.sns.subscribe(
+            TopicArn=self.TopicArn,
+            Protocol="sqs",
+            Endpoint=self.QueueArn
+        )
+
+        sub_arn = result["SubscriptionArn"]
+
+        self.sns.set_subscription_attributes(
+            SubscriptionArn=sub_arn,
+            AttributeName="RawMessageDelivery",
+            AttributeValue="true"
+        )
+        logging.debug('Done creating cloudsns resources')
+
+    @property
+    def TopicArn(self):
+        if not self.sqs or not self.sns:
+            raise NotInitialized('TopicArn')
+        if not self._topicArn:
+            topic = self.sns.create_topic(Name=self.topic_name)
+            self._topicArn = topic["TopicArn"]
+
+        return self._topicArn
+
+    @property
+    def QueueUrl(self):
+        if not self.sqs or not self.sns:
+            raise NotInitialized('QueueUrl')
+        if not self._queueUrl:
+            queue = self.sqs.create_queue(
+                QueueName=self.queue_name,
+                Attributes={"MessageRetentionPeriod": "60"}
+            )
+            self._queueUrl = queue["QueueUrl"]
+
+        return self._queueUrl
+
+    @property
+    def QueueArn(self):
+        if not self.sqs or not self.sns:
+            raise NotInitialized('QueueArn')
+        if not self._queueArn:
+            attr = self.sqs.get_queue_attributes(
+                QueueUrl=self.QueueUrl,
+                AttributeNames=["QueueArn"]
+            )
+            self._queueArn = attr["Attributes"]["QueueArn"]
+
+        return self._queueArn
+
+    def poll(self, user_fn):
+        completed = False
+        while not completed:
+            messages = self.get_messages()
+            for message in messages:
+                completed = user_fn(message)
+                message.delete()
+
+    def get_messages(self):
+        updates = self.sqs.receive_message(
+            QueueUrl=self.QueueUrl,
+            AttributeNames=["All"],
+            WaitTimeSeconds=20
+        )
+
+        if "Messages" in updates:
+            return [Message(m, self) for m in updates["Messages"]]
+
+        return []
+
+    def delete_messages(self, messages):
+        receipts = []
+        for m in messages:
+            unique_id = str(uuid.uuid4())
+            receipts.append({
+                'Id': unique_id,
+                'ReceiptHandle': m.ReceiptHandle
+            })
+
+        self.sqs.delete_message_batch(
+            QueueUrl=self.QueueUrl,
+            Entries=receipts
+        )
+
+    def cleanup(self, delete_topic=False):
+        logging.debug('Deleting cloudsns resources')
+
+        self.sqs.delete_queue(
+            QueueUrl=self.QueueUrl
+        )
+
+        if delete_topic:
+            self.sns.delete_topic(
+                TopicArn=self.TopicArn
+            )
+
+        logging.debug('Done deleting cloudsns resources')
```

Comment:
I'd also use double quotes around the log messages.  Again, use `logger`, not `logging`.  Make sure you update this throughout the file.